### ISSUE TITLE
Minimizing Data Leakage in Machine Learning

### ISSUE THREAD

**Issue by collins-a** [section-io/engineering-education#803]
### Brief Summary: 

We always want to produce the best performing machine learning models. If a model does well on our test set, we consider it to be a good model. Yet, a model may perform well during testing but poorly after deployment. There are a number of possible causes of such. One of them is data leakage. This refers to an error made by a creator of a model where information between test and training sets is accidentally shared. We explore this phenomenon in this article.  

### Key Takeaways: 

The reader should:

1. Understand data leakage and the motivation behind minimizing it
2. Understand the causes of data leakage
3. Learn effective techniques to minimize data leakage

### References: 

https://towardsdatascience.com/data-leakage-in-machine-learning-10bdd3eec742#:~:text=%20Data%20Leakage%20in%20Machine%20Learning%20%201,It%20is%20important%20to%20avoid%20these...%20More%20

https://machinelearningmastery.com/data-leakage-machine-learning/

https://towardsdatascience.com/data-leakage-in-machine-learning-6161c167e8ba

https://analyticsindiamag.com/what-is-data-leakage-in-ml-why-should-you-be-concerned/

https://mlinproduction.com/data-leakage/

https://www.datasciencecoffee.com/2020-data-leakage/

https://insights.ai-jobs.net/understanding-what-is-data-leakage-in-machine-learning-and-how-it-can-be-detected/

**Comment by ninjaginja** [comment]
great topic @collins-a - approved
